services/housekeeper/housekeeper.go

```diff
@@ -154,52 +119,103 @@ func (hk *Housekeeper) processNewSlot(headSlot uint64) {
 
 	// Update proposer duties
 	go hk.updateProposerDuties(headSlot)
-	go func() {
-		err := hk.redis.SetStats(datastore.RedisStatsFieldLatestSlot, headSlot)
-		if err != nil {
-			log.WithError(err).Error("failed to set stats")
-		}
-	}()
 
-	hk.headSlot.Store(headSlot)
+	// Set headSlot in redis (for the website)
+	err := hk.redis.SetStats(datastore.RedisStatsFieldLatestSlot, headSlot)
+	if err != nil {
+		log.WithError(err).Error("failed to set stats")
+	}
+
 	currentEpoch := headSlot / common.SlotsPerEpoch
 	log.WithFields(logrus.Fields{
 		"epoch":              currentEpoch,
 		"slotStartNextEpoch": (currentEpoch + 1) * common.SlotsPerEpoch,
 	}).Infof("updated headSlot to %d", headSlot)
 }
 
+// updateKnownValidators queries the full list of known validators from the beacon node
+// and stores it in redis. For the CL client this is an expensive operation and takes a bunch
+// of resources. This is why we schedule the requests for slot 4 and 20 of every epoch,
```

Comment:
why do we do this twice per epoch? once should be sufficient right ?

Reply:
i guess so, it was there since the beginning i think we did 2x/epoch just in case and cuz it didn't hurt. Do you think we should only do it in slot 4?